src/python/WMComponent/WorkflowUpdater/SiteListUpdater.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""
+File       : SiteListUpdater
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: provides module to manage update of site lists
+"""
+
+# system modules
+import os
+import json
+
+# WMCore modules
+from WMCore.Services.pycurl_manager import getdata
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from Utils.CertTools import ckey, cert
+
+class SiteListUpdater(object):
+    def __init__(self, url, logger):
+        """
+        SiteListUpdater provides codebase to update site list info in local agent.
+        Its logic is based on the following:
+        - obtain list of current active workflows from the agent
+        - requests their specs from upstream ReqMgr2 server
+        - update site lists of all workflows
+        - push new specs to the agent database
+
+        :logger: logger to use
+        """
+        self.url = url
+        self.logger = logger
+
+    def getActiveRequests(self, wflowSpecs):
```

Comment:
This should be using a SQL statement, similar to what you have implemented for pileup updates in WorkflowUpdater.

Reply:
If we'll put this class to be used in `WorkflowUpdaterPoller.py` (as I think it will be), then the `WorkflowUpdaterPoller.py` code does this SQL statement I didn't want to do it again as we already got wflowSpecs in here: https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py#L286